### ISSUE TITLE
datafeeder - source code is split between georchestra and geonetwork-ui

### ISSUE THREAD

**Issue by jeanpommier** [georchestra/georchestra#4066]
This makes it difficult to figure out how the app is wired. The datafeeder-ui is only encapsulating the frontend in a webapp, but its source code is at https://github.com/georchestra/geonetwork-ui/tree/master/apps/datafeeder
(and not documented at all, no build instructions, nothing I could see)

It would be preferrable to host all the apps' source code (backend and frontend) at the same place.

**Comment by jahow** [comment]
I agree, this has been a long-standing architecture quirk. Keep in mind also that, from the GeoNetwork-UI project, the Datafeeder application remains kind of a mystery because its backend is in a downstream project (geOrchestra) and not GeoNetwork itself.

What I'm pushing for on this topic is to use part of the future budget on the Datafeeder to somehow refactor it into a [GeoNetwork microservice](https://github.com/geonetwork/geonetwork-microservices), or "microservice" (notice the quotes). This would mean keeping some kind of plugin or implementation in geOrchestra side for the specifics; but really most of it could be agnostic to the infrastructure it's part of.

Having the Datafeeder be a citizen of the GN ecosystem would bring many benefits IMO, its features are often sought after for users of data catalogs.

Note that this requires the GN-microservices repository to also be reorganized to facilitate the use of its various modules.

**Comment by jeanpommier** [comment]
In the light of a discussion I just had with @fphg, I would actually suggest otherwise. 

If we make datafeeder a geonetwork microservice, it makes it very much tied to GN, hence to geospatial world.

Since we are considering enlarging geOrchestra to the data world (geo and non-geo), it might be more interesting to:
- make datafeeder a geo-agnostic project. It shouldn't even know about the end-user apps. Not tied with GN, GS, data api or whatever. Its purpose would be to push into an agnostic "data lake". It could be postgresql/postgis or something else, we don't really care, we just have to decide at some point, a "camp de base" as Fabrice puts it, that goes as far as we can go incommon (people from the geo world, people from the data/stats world)
- build a central, managing app, that is still agnostic, but care about where the data came from, where it is stored, who published it + some metrics that will allow to track change in the data (structure, number of items, bounding box etc). And might be able to warn about possible breaking changes that would impact downstream. 
- add post-publish hooks for every downstream business-specific apps we want to publish to (GN, GS, data api, etc) making easy to publish/unpublish. After publishing the data with the datafeeder (the basecamp), people could just trigger any of those hooks to go one step further, but now into business-logic apps.

That would make it much more flexible and open to further changes or addition of additional software.

**Comment by jeanpommier** [comment]
And we probably can make it so that it could also be used outside of geOrchestra environment for people that use upstream GN. But I would really prefer to decouple it as much as possible.

**Comment by fvanderbiest** [comment]
> If we make datafeeder a geonetwork microservice, it makes it very much tied to GN, hence to geospatial world.

Not 100% true: datahub targets the opendata world, with a GeoNetwork backend.

**Comment by fvanderbiest** [comment]
> Since we are considering enlarging geOrchestra to the data world (geo and non-geo), it might be more interesting to

The plan is great, but requires lots of funding that we currently do not have.
This would require another crowdfunding IMO.

**Comment by jeanpommier** [comment]
> The plan is great, but requires lots of funding that we currently do not have.

Hi @fvanderbiest 
I'm not sure this is the time and place to talk about funding. After all, this is an open source project, you never know what can occur. I believe it would be actually beneficial for all to take some time to sit around a (virtual) table and figure out what would be our ideal geOrchestra. The project shouldn't be solely driven by the customer's demand, because this would induce a lack of vision at large.

And actually, my proposal might not require so much work.

**Comment by fvanderbiest** [comment]
100% agree to sit around a table and sketch geOrchestra's future :-)

**Comment by f-necas** [comment]
Closing as datafeeder is now a pure georchestra product
